### ISSUE TITLE
Not all outputs have names

### ISSUE THREAD

**Issue by tobigue** [Ki6an/fastT5#5]
When inspecting the generated onnx models I saw that the decoder and init_decoder have a lot of unnamed outputs.

You can see the output names of an onnx InferenceSession like this: `list(map(lambda x: x.name, session.get_outputs()))`

We can see that the first self attention is called `past_key_values` while the other outputs have generated names.
![image](https://user-images.githubusercontent.com/1560152/113336407-a9199b00-9326-11eb-926d-c2339447276c.png)

I was just wondering if this can be a problem, if the [dynamic axes are not set for the other outputs](https://github.com/Ki6an/fastT5/blob/3555f11859f4a856986c80ec93a30076d078be34/fastT5/onnx_exporter.py#L214).

Also, could it be possible that the output_past_key_values dynamic axes for sequence should be `2` (as in `dyn_pkv` inputs) instead of `1`?

Cheers!

**Comment by Ki6an** [comment]
the onnx model does not produce the same output shape as the PyTorch model, the PyTorch model gives past_key_values as a nested tuple of tensors of shape (6x4). but the onnx model outputs everything as a single list. ( logits + 24 pkv tensors ) (the eg: shape here is of t5-small)

>I was just wondering if this can be a problem if the dynamic axes are not set for the other outputs.

I've tested it and it does not seem to cause any problem as it is generated by model but you can set the dynamic axis for output past_key_values as we did [here ](https://github.com/Ki6an/fastT5/blob/3555f11859f4a856986c80ec93a30076d078be34/fastT5/onnx_exporter.py#L158) for input past_key_values.

**Comment by tobigue** [comment]
Thanks for the answer. Yeah, I also could not find any information in the onnx documentation on whether it is important to set output dynamic axes or not.
